lib/transports/polling-xhr.js

```diff
@@ -185,6 +193,16 @@ Request.prototype.create = function(){
   try {
     debug('xhr open %s: %s', this.method, this.uri);
     xhr.open(this.method, this.uri, this.async);
+    try {
+      if (this.extraHeaders) {
+        xhr.setDisableHeaderCheck(true);
```

Comment:
See this comment ;-)

https://github.com/Automattic/engine.io-client/pull/379#issuecomment-85626842


That call (https://github.com/driverdan/node-XMLHttpRequest/blob/master/lib/XMLHttpRequest.js#L177) is needed so this call (https://github.com/driverdan/node-XMLHttpRequest/blob/master/lib/XMLHttpRequest.js#L192) does work correctly.